File: datasets/synthetic/SSTI-JA-REPO-002/src/main/java/com/example/demo/service/ReportService.java
package com.example.demo.service;

import com.example.demo.util.TemplateUtil;

import java.util.regex.Pattern;

public class ReportService {

    private static final Pattern ALLOWED_TITLE =
            Pattern.compile("^[A-Za-z0-9 _,.:;!?()-]{1,128}$");

    private String validateTitle(String title) {
        if (title == null) return null;
        String stripped = title.strip();
        return ALLOWED_TITLE.matcher(stripped).matches() ? stripped : null;
    }

    public String renderPreview(String title) {
        String safeTitle = validateTitle(title);
        if (safeTitle == null) {
            return null;
        }
        return TemplateUtil.renderReportTemplate(safeTitle);
    }
}

File: datasets/synthetic/SSTI-PY-REPO-002/src/services/report_service.py
import re

from utils.template_utils import render_report_template

_ALLOWED_TITLE = re.compile(r'^[A-Za-z0-9 _,.:;!?()-]{1,128}$')


class ReportService:
    def _validate_title(self, title: str) -> str | None:
        stripped = title.strip()
        if _ALLOWED_TITLE.fullmatch(stripped):
            return stripped
        return None

    def render_preview(self, title: str) -> str | None:
        safe_title = self._validate_title(title)
        if safe_title is None:
            return None
        return render_report_template(safe_title)
